### PR TITLE
feat: implement bottom nav using AutoTabsScaffold

### LINKED ISSUE #14: Update the bottom nav implementation to `AutoTabScaffold`
`AutoTabScaffold` is a widget from `auto_route` to make the bottom nav and keep the state the same in the view

- Create a screen named `MainScreen`
- Create 3 tabs under `MainScreen`
- 3 inner tabs should be named `~Page`

This means:
- `~Screen`s are navigatable and `~Page`s are non navigatable
- We used this naming pattern in the last 2 projects

### PR DESCRIPTION
**Pull request by ashraful-chowdhury**
# Pull Request (PR) Checklist

- [x] PR title includes Jira ticket reference -- if no ticket available create one if this PR is a fix or feat
- [x] Branch name includes Jira ticket reference (parent ticket and sub-task if available), e.g. 'feat/APP-XXX_Some_Feature' or feat/APP-XXX_APP-XXX_Some_Sub_Task or 'fix/APP-XXX_Some_Fix' or 'refactor/Some_Refactoring'
- [x] All PR checks pass and changes under /domain are covered with unit tests
- [x] Jira ticket provides testing infos for QA -- add a comment if needed e.q. for pre-requisites, order of API calls, ...
- [x] When merging this PR the commit message follows [conventional-commits](https://www.conventionalcommits.org/en/v1.0.0/#summary), e.g. 'feat(some optional epic): add some feature (APP-XXX) (#XXX)'

### CODE REVIEW

File: lib/presentation/feature/profile/cubit/profile_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import 'package:injectable/injectable.dart';
+
+@Injectable()
+class ProfileCubit extends Cubit<ProfileState> {
+  final GetProfileNameUseCase _getProfileNameUseCase;
+
+  ProfileCubit(this._getProfileNameUseCase) : super(ProfileState.initial());
+
+  Future<void> load() async {
+    if (state.status == const BaseStatus.loading()) return;
```
Comment (utpal-barman) [inline lib/presentation/feature/profile/cubit/profile_cubit.dart:16]: use extension
state.status.isLoading

File: lib/presentation/feature/profile/cubit/profile_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          ),
+        )
+        .catchPrintError((e, s) {
+      emit(
+        state.copyWith(
+          status: BaseStatus.failure(ResponseError.from(e)),
+          name: '',
+        ),
+      );
+    });
```
Comment (utpal-barman) [inline lib/presentation/feature/profile/cubit/profile_cubit.dart:37]: use try catch with async-await

File: lib/presentation/feature/profile/ui/profile_body.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        child: BlocBuilder<ProfileCubit, ProfileState>(
+          builder: (context, state) {
+            debugPrint(state.toString());
+            if (state.status.isSuccess) {
+              return Text('Hi ${state.name}');
+            } else if (state.status.isFailure) {
+              return const Text('Something went wrong!');
+            }
+            return const CircularProgressIndicator();
+          },
```
Comment (utpal-barman) [inline lib/presentation/feature/profile/ui/profile_body.dart:25]: use when or null for conditional based rendering

File: lib/presentation/feature/splash/splash_screen.dart [hunk 4]
```diff
@@ -19,7 +19,7 @@ class _SplashScreenState extends State<SplashScreen> {
   @override
   void initState() {
     _timer = Timer(const Duration(seconds: 2), () {
-      context.router.navigate(HomeRoute());
+      context.router.replace(const MainRoute());
```
Comment (utpal-barman) [inline lib/presentation/feature/splash/splash_screen.dart:22]: use from build context when possible
```dart
context.pushRoute(...)
context.popRoute(...)
```

**Review by utpal-barman — CHANGES_REQUESTED** [review]
change the pubspect

```yaml
environment:
  sdk: ">=3.0.0 <4.0.0"
```

File: lib/presentation/feature/splash/splash_screen.dart (hunk 4 above)
Comment (ashraful-chowdhury) [reply lib/presentation/feature/splash/splash_screen.dart:22]: bhai, not getting the point, would you please elaborate a bit please

File: lib/presentation/feature/profile/cubit/profile_cubit.dart [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        status: const BaseStatus.success(),
+        name: result ?? '',
+      ));
+    } catch (e) {
+      if (isClosed) {
+        return;
+      }
+      emit(state.copyWith(
+        status: BaseStatus.failure(ResponseError.from(e)),
+        name: '',
```
Comment (utpal-barman) [inline lib/presentation/feature/profile/cubit/profile_cubit.dart:34]: `name: ''`  not required

**Review by utpal-barman — APPROVED** [review]
Just one more small req, otherwise good to go

File: lib/presentation/feature/profile/cubit/profile_cubit.dart (hunk 5 above)
Comment (ashraful-chowdhury) [reply lib/presentation/feature/profile/cubit/profile_cubit.dart:34]: bhai, how to handle nullable values then?

**Pull request merged**